graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -433,11 +506,11 @@ abstract class BaseDataTypeGenerator(
 
     private fun addBuilder(javaType: TypeSpec.Builder) {
         val className = ClassName.get(packageName, javaType.build().name)
-        val buildMethod = MethodSpec.methodBuilder("build").returns(className).addStatement(
+        var buildMethod = MethodSpec.methodBuilder("build").returns(className).addCode(
```

Comment:
in general with kotlin we should prefer the use of `val`. Since `if` statements are expressions we could probably achieve the same logic with a statement such as

```
val buildMethod = if(config.generateBitset) { 
   ...
} else {
 ...
}
```

or even better, it might be possible to put this in its own method

```
private fun createBuildMethod() {
  return if (config.generateBitset) {
    ...
  } else {
   ...
  }
}
```